tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ExtraEntrySortedMap.java

```diff
@@ -0,0 +1,207 @@
+        }
+        return ordering.max(base.lastKey(), extraKey);
+    }
+
+    @Override
+    public int size() {
+        return base.size() + 1;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return false;
+    }
+
+    @Override
+    public boolean containsKey(Object key) {
+        return extraKey.equals(key) || base.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(Object value) {
+        return extraValue.equals(value) || base.containsValue(value);
+    }
+
+    @Override
+    public V get(Object key) {
+        if (extraKey.equals(key)) {
+            return extraValue;
+        }
+        return base.get(key);
+    }
+
+    @Override
+    public V put(K key, V value) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public V remove(Object key) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void putAll(Map<? extends K, ? extends V> other) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void clear() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Set<K> keySet() {
+        return new AbstractSet<K>() {
+            @Override
+            public Iterator<K> iterator() {
+                return Iterables.mergeSorted(ImmutableList.of(base.keySet(), ImmutableList.of(extraKey)), ordering)
+                        .iterator();
+            }
+
+            @Override
+            public int size() {
+                return base.size() + 1;
+            }
+        };
+    }
+
+    @Override
+    public Collection<V> values() {
+        return new AbstractCollection<V>() {
+            @Override
+            public Iterator<V> iterator() {
+                return keySet().stream().map(ExtraEntrySortedMap.this::get).iterator();
+            }
+
+            @Override
+            public int size() {
+                return base.values().size() + 1;
+            }
+        };
+    }
+
+    @Override
+    public Set<Entry<K, V>> entrySet() {
+        return new AbstractSet<Map.Entry<K, V>>() {
+            @Override
+            public Iterator<Map.Entry<K, V>> iterator() {
+                return Iterators.transform(keySet().iterator(), key -> Maps.immutableEntry(key, get(key)));
+            }
+
+            @Override
+            public int size() {
+                return base.size() + 1;
+            }
+        };
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other instanceof ExtraEntrySortedMap) {
+            ExtraEntrySortedMap otherMap = (ExtraEntrySortedMap) other;
+            if (extraKey.equals(otherMap.extraKey) && extraValue.equals(otherMap.extraValue)) {
+                return base.equals(otherMap.base);
+            }
+        }
+        return super.equals(other);
+    }
+
+    @Override
+    public int hashCode() {
+        return base.hashCode() + extraEntryHashCode;
```

Comment:
do we need to spread the hash more effectively given `String`'s poor hashCode?

Reply:
`Map` defines the contract of `Map.hashCode`